frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditDataType/EditDataType.test.tsx

```diff
@@ -0,0 +1,162 @@
+import type { BpmnApiContextProps } from '../../../../contexts/BpmnApiContext';
+import { BpmnApiContext } from '../../../../contexts/BpmnApiContext';
+import type { BpmnContextProps } from '../../../../contexts/BpmnContext';
+import { BpmnContext } from '../../../../contexts/BpmnContext';
+import type { BpmnDetails } from '../../../../types/BpmnDetails';
+import { BpmnTypeEnum } from '../../../../enum/BpmnTypeEnum';
+import type Modeler from 'bpmn-js/lib/Modeler';
+import { EditDataType } from './EditDataType';
+import { BpmnConfigPanelFormContextProvider } from '../../../../contexts/BpmnConfigPanelContext';
+
+const mockTaskId: string = 'testId';
+const mockName: string = 'testName';
+
+const modelerRefMock = {
+  current: {
+    get: () => {},
+  } as unknown as Modeler,
+};
+
+const mockBpmnDetails: BpmnDetails = {
+  id: mockTaskId,
+  name: mockName,
+  taskType: 'data',
+  type: BpmnTypeEnum.Task,
+};
+
+const mockBpmnApiContextValue: Partial<BpmnApiContextProps> = {
+  layoutSets: {
+    sets: [
+      {
+        id: 'layoutSetName',
+        tasks: [mockTaskId],
+      },
+    ],
+  },
+  availableDataModelIds: [],
+};
+
+const mockBpmnContextValue: Partial<BpmnContextProps> = {
+  bpmnDetails: mockBpmnDetails,
+  modelerRef: modelerRefMock,
+};
+
+describe('EditDataType', () => {
+  it('should display a button to add datamodel when task has no datamodel', () => {
+    renderEditDataType();
+    screen.getByRole('button', {
+      name: textMock('process_editor.configuration_panel_set_datamodel_link'),
+    });
+  });
+
+  it('should display a native select with default value when clicking "add datamodel"', async () => {
+    const user = userEvent.setup();
+    renderEditDataType();
+    const addDataModelButton = screen.getByRole('button', {
+      name: textMock('process_editor.configuration_panel_set_datamodel_link'),
+    });
+    await user.click(addDataModelButton);
+    const nativeSelect = screen.getByRole('combobox', {
+      name: textMock('process_editor.configuration_panel_set_datamodel'),
+    });
+    expect(nativeSelect).toHaveValue('noModelKey');
+  });
+
+  it('should display all available data types including existing and no-model-key as options for data type select', async () => {
+    const user = userEvent.setup();
+    const availableDataModelIds = ['dataModel1', 'dataModel2'];
+    const existingDataType = 'dataModel0';
+    renderEditDataType({
+      layoutSets: {
+        sets: [{ id: 'setWithDataType', dataType: existingDataType, tasks: [mockTaskId] }],
+      },
+      availableDataModelIds,
+    });
+    const updateDataTypeButton = screen.getByRole('button', {
+      name: textMock('process_editor.configuration_panel_set_datamodel'),
+    });
+    await user.click(updateDataTypeButton);
+
+    screen.getByRole('option', {
+      name: textMock('process_editor.configuration_panel_select_datamodel'),
+    });
+    screen.getByRole('option', { name: existingDataType });
```

Comment:
The Author of React Testing Libary has written an article about the [common mistakes when writing tests](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions). Where he points out that get* queries should be wrapped within expect. This is opinionated thing, but I do agree with him. Please read the text in the image below or [this link](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions) for more information. 

We should establish this a standard for our team, to ensure we are asserting things elements the same way cross the codebase.

![image](https://github.com/Altinn/altinn-studio/assets/46874830/7ec7af73-db05-476b-a433-71a6b2a2c8b9)